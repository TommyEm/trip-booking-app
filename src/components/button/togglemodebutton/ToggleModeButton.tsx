import React, {
	useCallback,
	useContext,
	useState,
} from 'react';
import { ToggleButton } from 'react-bootstrap';

import { StyledToggleModeButton } from './ToggleModeButton.styled';
import * as actions from '../../../store/actionTypes';
import { AppContext } from '../../../store/Store';


export interface IToggleModeButtonProps {
	className?: string;
}

export const ToggleModeButton: React.FC<IToggleModeButtonProps> = ({
	className,
}: IToggleModeButtonProps) => {

	const [isDarkMode, setDarkMode] = useState<boolean>(true);
	const { dispatch } = useContext(AppContext);

	const toggleThemeMode = useCallback(() => {
		setDarkMode(!isDarkMode);
		dispatch({ type: actions.UPDATE_THEME_MODE, payload: !isDarkMode });
	}, [dispatch, isDarkMode]);

    return (
		<StyledToggleModeButton
			className={`ToggleModeButton ${isDarkMode ? 'mod-checked' : ''} ${className}`}
			data-testid='test-ToggleModeButton'
		>
			<span>Light</span>
			<ToggleButton
				id='toggle-dark-mode'
				checked={isDarkMode}
				value='dark'
				onClick={toggleThemeMode}
			>
				Toggle
			</ToggleButton>
			<span>Dark</span>
		</StyledToggleModeButton>
	);
}
