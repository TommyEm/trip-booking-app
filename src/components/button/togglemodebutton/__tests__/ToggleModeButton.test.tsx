import React from 'react';
import {
	cleanup,
	render,
	screen
} from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';

import * as stories from '../ToggleModeButton.stories';

const { Default } = composeStories(stories);


afterEach(cleanup);
const storyComponent = <Default />;

it('should render ...', () => {
	render(storyComponent);
	expect(screen.getByTestId('test-ToggleModeButton')).toBeVisible();
});

it('matches snapshot', () => {
	const tree = render(storyComponent);
	expect(tree).toMatchSnapshot();
});
