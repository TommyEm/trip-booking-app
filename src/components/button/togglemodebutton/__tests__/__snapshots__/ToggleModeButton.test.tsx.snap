// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdvvtL jUgbOi ToggleModeButton mod-checked undefined"
        data-testid="test-ToggleModeButton"
      >
        <span>
          Light
        </span>
        <input
          autocomplete="off"
          checked=""
          class="btn-check"
          id="toggle-dark-mode"
          value="dark"
        />
        <label
          class="btn btn-primary"
          for="toggle-dark-mode"
          role="button"
          tabindex="0"
        >
          Toggle
        </label>
        <span>
          Dark
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdvvtL jUgbOi ToggleModeButton mod-checked undefined"
      data-testid="test-ToggleModeButton"
    >
      <span>
        Light
      </span>
      <input
        autocomplete="off"
        checked=""
        class="btn-check"
        id="toggle-dark-mode"
        value="dark"
      />
      <label
        class="btn btn-primary"
        for="toggle-dark-mode"
        role="button"
        tabindex="0"
      >
        Toggle
      </label>
      <span>
        Dark
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
